src/stk/common/log.h

```diff
@@ -139,3 +148,8 @@ stk::LogMessage& operator<<(stk::LogMessage& s, const T& v)
 #endif
 
 #define LOG_IF(level, expr) !(expr) ? (void)0 : LOG(level)
+
+// Evaluate the expression to be logged lazily, only if there is at least
+// one sink that will consume it.
+#define LOG_LAZY(level, expr) if (stk::level >= stk::log_level()) { LOG(level) << expr; }
```

Comment:
I think it would make sense to just add this to the regular `LOG` macro. Having a `LOG` and a `LOG_LAZY` is just confusing.

Reply:
I agree. It is just it involves a lot of replacements along the code, from `LOG(Level) << stuff;` to `LOG(Level, stuff)`, so I didn't go for it at first. But if it is ok for you, I can easily make the change.